aquadoggo/src/domain.rs

```diff
@@ -202,44 +208,60 @@ pub async fn next_args<S: StorageProvider>(
 /// ## Compute and return next entry arguments
 pub async fn publish<S: StorageProvider>(
     store: &S,
-    entry_encoded: &EntrySigned,
-    operation_encoded: &OperationEncoded,
+    schema: &Schema,
+    encoded_entry: &EncodedEntry,
+    plain_operation: &PlainOperation,
+    encoded_operation: &EncodedOperation,
 ) -> Result<NextEntryArguments> {
-    ////////////////////////////////
-    // DECODE ENTRY AND OPERATION //
-    ////////////////////////////////
+    //////////////////
+    // DECODE ENTRY //
+    //////////////////
 
-    let entry = decode_entry(entry_encoded, Some(operation_encoded))?;
-    let operation = Operation::from(operation_encoded);
-    let author = entry_encoded.author();
+    let entry = decode_entry(encoded_entry)?;
+    let author = entry.public_key();
     let log_id = entry.log_id();
     let seq_num = entry.seq_num();
 
-    ///////////////////////////
-    // VALIDATE ENTRY VALUES //
-    ///////////////////////////
+    //////////////////////////////////
+    // VALIDATE ENTRY AND OPERATION //
+    //////////////////////////////////
+
+    // @TODO: Check this validation flow is still correct.
 
     // Verify that the claimed seq num matches the expected seq num for this author and log.
     let latest_entry = store.get_latest_entry(&author, log_id).await?;
     let latest_seq_num = latest_entry.as_ref().map(|entry| entry.seq_num());
-    is_next_seq_num(latest_seq_num.as_ref(), seq_num)?;
+    is_next_seq_num(latest_seq_num, seq_num)?;
 
     // The backlink for this entry is the latest entry from this public key's log.
     let backlink = latest_entry;
 
     // If a skiplink is claimed, get the expected skiplink from the database, errors
     // if it can't be found.
-    let skiplink = match entry.skiplink_hash() {
+    let skiplink = match entry.skiplink() {
         Some(_) => Some(get_expected_skiplink(store, &author, log_id, seq_num).await?),
         None => None,
     };
```

Comment:
This flow seems to not match now. What is currently happening is that we fetch the `expected` skiplink based on the author and seq num encoded on the passed entry. Would we rather just fetch the claimed skiplink based on the encoded skiplink hash?

Reply:
Why does it not match?

Reply:
Well, as I understand it, in [`validate_log_integrity()`](https://github.com/p2panda/p2panda/blob/53e74101c00022e46c35093a78f049431b41376d/p2panda-rs/src/entry/validate.rs#L48) we are expecting the claimed skiplink rather than the expected.

Reply:
Either way the validation works, but maybe the flow would be in a different order depending on the two approaches?